### ISSUE TITLE
[ feature request ] use gitoxide

### ISSUE THREAD

**Issue by yonas** [ndr3www/git-conform#3]
Remove dependency on external `git` command by using [gitoxide](https://github.com/GitoxideLabs/gitoxide).

Gitoxide is an idiomatic, lean, fast & safe pure Rust implementation of Git.

**Comment by j3dr** [comment]
Yeah, I've been thinking about ditching `git` commands lately and using some crate to do the work instead. Might happen in the next major release.
